### ISSUE TITLE
Current PF_RING from Git kills OpenVZ kernel if called for "any" interface

### ISSUE THREAD

**Issue by pavel-odintsov** [ntop/PF_RING#41]
Hello!

Have a stable reproducer. Start vanilla PF_RING for "any" from Git on OpenVZ server and kill it:

``` bash
evo0.fv.ee:  CR2: 00000000000000a0
evo0.fv.ee:   RSP <ffff881045afbca8>
evo0.fv.ee:  [<ffffffffa0a56080>] ring_getsockopt+0x5a0/0xdf0 [pf_ring]
evo0.fv.ee: ff
evo0.fv.ee:  RIP
evo0.fv.ee: 8c
evo0.fv.ee: fd
evo0.fv.ee: ff
evo0.fv.ee: ff
evo0.fv.ee: e9
evo0.fv.ee: 68
evo0.fv.ee: fe
evo0.fv.ee: c2
evo0.fv.ee: 88
evo0.fv.ee: 85
evo0.fv.ee: d1
evo0.fv.ee: 0f
evo0.fv.ee: 46
evo0.fv.ee: 00
evo0.fv.ee: 00
evo0.fv.ee: 39
evo0.fv.ee: 92
evo0.fv.ee: a0
evo0.fv.ee: 00
evo0.fv.ee: c8
evo0.fv.ee: <8b>
evo0.fv.ee: e0
evo0.fv.ee: 0f
evo0.fv.ee: b6
evo0.fv.ee: 8b
evo0.fv.ee: 52
evo0.fv.ee: 43
evo0.fv.ee: 02
evo0.fv.ee: 48
evo0.fv.ee: 28
evo0.fv.ee: 0f
evo0.fv.ee: b6
evo0.fv.ee: 48
evo0.fv.ee: 8b
evo0.fv.ee: 52
evo0.fv.ee: ff
evo0.fv.ee: ff
evo0.fv.ee: e9
evo0.fv.ee: 8e
evo0.fv.ee: fe
evo0.fv.ee: ee
evo0.fv.ee: 80
evo0.fv.ee: 28
evo0.fv.ee: 48
evo0.fv.ee: 83
evo0.fv.ee: 48
evo0.fv.ee: 8b
evo0.fv.ee: 72
evo0.fv.ee: ff
evo0.fv.ee: ff
evo0.fv.ee: e9
evo0.fv.ee: c3
evo0.fv.ee: fb
evo0.fv.ee: ff
evo0.fv.ee: ff
evo0.fv.ee: 12
evo0.fv.ee: fb
evo0.fv.ee: 0f
evo0.fv.ee: 85
evo0.fv.ee: 48
evo0.fv.ee: 85
evo0.fv.ee: c0
evo0.fv.ee:  Code:
evo0.fv.ee: e0
evo0.fv.ee:   [<ffffffff8100b1a2>] system_call_fastpath+0x16/0x1b
evo0.fv.ee:   [<ffffffff8145b2c5>] sys_getsockopt+0x65/0xc0
evo0.fv.ee:   [<ffffffffa0093008>] ? ext4_file_write+0x58/0x190 [ext4]
evo0.fv.ee:   [<ffffffff8114239a>] ? generic_file_aio_write+0xba/0x100
evo0.fv.ee:   [<ffffffff811b819e>] ? __sb_end_write+0x3e/0x90
evo0.fv.ee:   [<ffffffff811422c5>] ? __generic_file_aio_write+0x85/0xa0
evo0.fv.ee:   [<ffffffff81472575>] ? __dev_get_by_name+0x25/0xd0
evo0.fv.ee:   [<ffffffff81142045>] ? __generic_file_write_iter+0x225/0x420
evo0.fv.ee:  Call Trace:
evo0.fv.ee:  000000000000003a
evo0.fv.ee:  ffff88101c220a80
evo0.fv.ee:  ffff880392048cd0
evo0.fv.ee:  <d>
evo0.fv.ee:  000000000000003a
evo0.fv.ee:  00000000000049f3
evo0.fv.ee:  ffffea002c0f2a80
evo0.fv.ee:  ffff8805e5388fc0
evo0.fv.ee:  <d>
evo0.fv.ee:  0000000000000000
evo0.fv.ee:  ffff880392048cd0
evo0.fv.ee:  000000000000060d
evo0.fv.ee:  000000000000003a
evo0.fv.ee:  Stack:
evo0.fv.ee:   ffff88101c220a80
evo0.fv.ee:  Process fastnetmon (pid: 880814, veid: 0, threadinfo ffff881045afa000, task ffff8805e5388fc0)
evo0.fv.ee:  DR3: 0000000000000000 DR6: 00000000ffff0ff0 DR7: 0000000000000400
evo0.fv.ee:  DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
evo0.fv.ee:  CR2: 00000000000000a0 CR3: 0000000ff286f000 CR4: 00000000001407e0
evo0.fv.ee:  CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
evo0.fv.ee:  FS:  00007f5e90c0d700(0000) GS:ffff8800689c0000(0000) knlGS:0000000000000000
evo0.fv.ee:  R13: 00007f5e90c0cbe8 R14: 00007f5e90c0cbef R15: 0000000000000004
evo0.fv.ee:  R10: 00007f5e90c0cbef R11: 0000000000000246 R12: 00000000000000ae
evo0.fv.ee:  RBP: ffff881045afbf28 R08: 00007f5e90c0cbe8 R09: 203237342c35343a
evo0.fv.ee:  RDX: 0000000000000000 RSI: 0000000000000000 RDI: ffff88101d866980
evo0.fv.ee:  RAX: 0000000000000000 RBX: ffff88100623d800 RCX: 0000000000000000
evo0.fv.ee:  RSP: 0018:ffff881045afbca8  EFLAGS: 00010246
evo0.fv.ee:  [<ffffffffa0a56080>] ring_getsockopt+0x5a0/0xdf0 [pf_ring]
evo0.fv.ee:  RIP: 0010:[<ffffffffa0a56080>]
evo0.fv.ee:  Supermicro X9SRD-F
evo0.fv.ee: /X9SRD-F
evo0.fv.ee:  Pid: 880814, comm: fastnetmon veid: 0 Tainted: P           ---------------    2.6.32-042stab111.11 #1 042stab111_11
evo0.fv.ee:  dm_mod
evo0.fv.ee:  [last unloaded: pf_ring]
evo0.fv.ee:  dm_mirror
evo0.fv.ee:  dm_region_hash
evo0.fv.ee:  dm_log
evo0.fv.ee:  ahci
evo0.fv.ee:  wmi
evo0.fv.ee:  sd_mod
evo0.fv.ee:  crc_t10dif
evo0.fv.ee:  mbcache
evo0.fv.ee:  raid1
evo0.fv.ee:  shpchp
evo0.fv.ee:  ext4
evo0.fv.ee:  jbd2
evo0.fv.ee:  lpc_ich
evo0.fv.ee:  mfd_core
evo0.fv.ee:  i2c_i801
evo0.fv.ee:  i2c_core
evo0.fv.ee:  sb_edac
evo0.fv.ee:  edac_core
evo0.fv.ee:  ptp
evo0.fv.ee:  pps_core
evo0.fv.ee:  dca
evo0.fv.ee:  i2c_algo_bit
evo0.fv.ee:  iTCO_vendor_support
evo0.fv.ee:  igb
evo0.fv.ee:  iTCO_wdt
evo0.fv.ee:  acpi_pad
evo0.fv.ee:  serio_raw
evo0.fv.ee:  ipmi_msghandler
evo0.fv.ee:  sg
evo0.fv.ee:  nf_defrag_ipv4
evo0.fv.ee:  tun
evo0.fv.ee:  ipmi_si
evo0.fv.ee:  nf_conntrack_ipv4
evo0.fv.ee:  nf_conntrack
evo0.fv.ee:  nf_nat
evo0.fv.ee:  ip_tables
evo0.fv.ee:  fuse
evo0.fv.ee:  iptable_nat
evo0.fv.ee:  slhc
evo0.fv.ee:  crc_ccitt
evo0.fv.ee:  zlib_deflate
evo0.fv.ee:  ppp_async
evo0.fv.ee:  ppp_generic
evo0.fv.ee:  ppp_mppe
evo0.fv.ee:  ppp_deflate
evo0.fv.ee:  arc4
evo0.fv.ee:  ecb
evo0.fv.ee:  esp4
evo0.fv.ee:  af_key
evo0.fv.ee:  esp6
evo0.fv.ee:  ipv6
evo0.fv.ee:  pppox
evo0.fv.ee:  xfrm6_mode_tunnel
evo0.fv.ee:  xfrm4_mode_tunnel
evo0.fv.ee:  xfrm4_mode_transport
evo0.fv.ee:  pppol2tp
evo0.fv.ee:  ipt_LOG
evo0.fv.ee:  xfrm_ipcomp
evo0.fv.ee:  nf_defrag_ipv6
evo0.fv.ee:  xt_state
evo0.fv.ee:  xt_owner
evo0.fv.ee:  nf_conntrack_ipv6
evo0.fv.ee:  xt_connlimit
evo0.fv.ee:  ipt_REDIRECT
evo0.fv.ee:  iptable_raw
evo0.fv.ee:  iptable_mangle
evo0.fv.ee:  ipt_MASQUERADE
evo0.fv.ee:  nfs_acl
evo0.fv.ee:  sunrpc
evo0.fv.ee:  fscache
evo0.fv.ee:  auth_rpcgss
evo0.fv.ee:  nfs
evo0.fv.ee:  lockd
evo0.fv.ee:  vzmon
evo0.fv.ee:  vzdev
evo0.fv.ee:  vzrst
evo0.fv.ee:  vzcpt
evo0.fv.ee:  nf_conntrack_ftp
evo0.fv.ee:  xt_recent
evo0.fv.ee:  ipip
evo0.fv.ee:  tunnel4
evo0.fv.ee: (U)
evo0.fv.ee:  zunicode(P)
evo0.fv.ee: (U)
evo0.fv.ee: (U)
evo0.fv.ee:  zavl(P)
evo0.fv.ee:  znvpair(P)
evo0.fv.ee: (U)
evo0.fv.ee:  spl
evo0.fv.ee: (U)
evo0.fv.ee: (U)
evo0.fv.ee:  zcommon(P)
evo0.fv.ee:  ip6table_filter
evo0.fv.ee:  ip6_tables
evo0.fv.ee:  zfs(P)
evo0.fv.ee:  iptable_filter
evo0.fv.ee:  vznetdev
evo0.fv.ee:  bonding
evo0.fv.ee:  garp
evo0.fv.ee:  stp
evo0.fv.ee:  llc
evo0.fv.ee:  configfs
evo0.fv.ee:  8021q
evo0.fv.ee:  vzevent
evo0.fv.ee:  netconsole
evo0.fv.ee:  ip6t_REJECT
evo0.fv.ee:  ip6table_mangle
evo0.fv.ee:  vziolimit
evo0.fv.ee:  vzdquota
evo0.fv.ee:  ploop
evo0.fv.ee:  simfs
evo0.fv.ee:  pfmt_raw
evo0.fv.ee:  pfmt_ploop1
evo0.fv.ee:  pio_nfs
evo0.fv.ee:  pio_direct
evo0.fv.ee:  vzethdev
evo0.fv.ee:  pio_kaio
evo0.fv.ee:  ipt_REJECT
evo0.fv.ee:  xt_conntrack
evo0.fv.ee:  xt_multiport
evo0.fv.ee:  coretemp
evo0.fv.ee:  xt_TCPMSS
evo0.fv.ee:  freq_table
evo0.fv.ee:  mperf
evo0.fv.ee:  cpufreq_ondemand
evo0.fv.ee:  acpi_cpufreq
evo0.fv.ee:  cls_u32
evo0.fv.ee:  sch_htb
evo0.fv.ee:  kcare
evo0.fv.ee: (U)
evo0.fv.ee:  sch_sfq
evo0.fv.ee:  pf_ring
evo0.fv.ee: (U)
evo0.fv.ee:  Modules linked in:
evo0.fv.ee:  CPU 7
evo0.fv.ee:  last sysfs file: /sys/devices/system/node/node0/meminfo
evo0.fv.ee:  Oops: 0000 [#1]
evo0.fv.ee: SMP
evo0.fv.ee: PMD 0
evo0.fv.ee:  PGD f5d22b067
evo0.fv.ee: PUD 105f347067
evo0.fv.ee:  [<ffffffffa0a56080>] ring_getsockopt+0x5a0/0xdf0 [pf_ring]
evo0.fv.ee:  IP:
evo0.fv.ee: NULL pointer dereference
evo0.fv.ee:  at 00000000000000a0
evo0.fv.ee:  BUG: unable to handle kernel
```

Kernel version: 2.6.32-042stab111.11

**Comment by pavel-odintsov** [comment]
Any news?

**Comment by cardigliano** [comment]
Not yet, we do not have an OpenVZ testbed ready at the moment, we need to setup it.

**Comment by pavel-odintsov** [comment]
We could deploy server for you with OpenVZ. Is it OK?

**Comment by cardigliano** [comment]
Yes thank you

**Comment by pavel-odintsov** [comment]
We will prepare server in next few days and I will email you.

**Comment by cardigliano** [comment]
Fixed with 863a7342389117a897a754f6ab1c9e614630fd74, please double check.
Thank you for providing access to the server.

**Comment by pavel-odintsov** [comment]
Awesome! Thank you so much!
